### DISCUSSION TITLE
Question Regarding JoinLayer after Merging seurat object

### DISCUSSION THREAD

**Discussion by ylin575** [satijalab/seurat#8155] [Q&A] [answered]
Hello,

I had an issue with JoinLayer

The instruction for merging on the seurat website is :   merged_obj[["RNA"]] <- JoinLayers(merged_obj)
I had an error following this code:    Error in `[<-.data.frame`(`*tmp*`, , i, value = new("Seurat", assays = list( : 
     replacement has 39655 rows, data has 32128

I modified it to this:    merged_obj[["RNA"]] <- JoinLayers(merged_obj[["RNA"]])
which worked.

I want to ask if it is ok to use the modified code and proceed with my analysis.

Thanks,
Yong

============================

Instructions from seurat website:
   ### Merge two Seurat objects
   merged_obj <- merge(x = ifnb_list$CTRL, y = ifnb_list$STIM)
   merged_obj[["RNA"]] <- JoinLayers(merged_obj)

   ### Example to merge more than two Seurat objects
   merge(x = pbmc1, y = list(pbmc2, pbmc3))

### My code (returns an error):
   > hthy2[["RNA"]] <- JoinLayers(hthy2)
   Error in `[<-.data.frame`(`*tmp*`, , i, value = new("Seurat", assays = list( : 
     replacement has 39655 rows, data has 32128

My modified code (seem to work, did not return error):
   > hthy2[["RNA"]] <- JoinLayers(hthy2[["RNA"]])

**Comment by evgeny-bakin** [comment]
I have the same problem. Did you have any new insights?

**Comment by JaredBCollins** [comment, marked as answer]
I've had the same issue following the same tutorial, and resolved it the same way.

I think the  ["Seurat Command List"](https://satijalab.org/seurat/articles/essential_commands.html#:~:text=Merge%20objects%20(without%20integration),rejoin%20the%20layers%20after%20merging.) page may have outdated/incorrect commands. The JoinLayers command is given as you have modified it on the ["Seurat V5 Command Cheat Sheet"](https://satijalab.org/seurat/articles/seurat5_essential_commands) page. They were both committed on the same day, however, so I'm not sure. A dev may be able to explain reason for the different documentations better.

**Reply by ylin575** [reply]
Thanks for pointing me to the cheat sheet. It does look like the "Seurat Command List" page has a typo for the join layer.
